service/controller/app/v1/resource/status/create.go

```diff
@@ -50,11 +50,15 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	if !equals(desiredStatus, key.AppStatus(cr)) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for app %#q", name))
 
-		crCopy := cr.DeepCopy()
-		crCopy.ResourceVersion = chart.GetResourceVersion()
-		crCopy.Status = desiredStatus
+		// Get app CR again to ensure the resource version is correct.
+		currentCR, err := r.g8sClient.ApplicationV1alpha1().Apps(cr.Namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
```

Comment:
Fix is just to get the latest app CR. Also aligns it with the status resource in chart-operator.